### PR TITLE
snowflake-connector-fix

### PR DESCRIPTION
**Pull request by codingwithabhi**
### Describe your changes :

I worked on the fixing snowflake connector.
#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.

#
### Reviewers
@harshach @ayush-shah @pmbrull

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/snowflake.py [hunk 1]
```diff
@@ -62,23 +65,31 @@ def get_connection_url(self):
...
-            p_key = serialization.load_pem_private_key(
-                bytes(private_key, "utf-8"),
-                password=os.environ["SNOWFLAKE_PRIVATE_KEY_PASSPHRASE"].encode(),
-                backend=default_backend(),
-            )
-            pkb = p_key.private_bytes(
-                encoding=serialization.Encoding.DER,
-                format=serialization.PrivateFormat.PKCS8,
-                encryption_algorithm=serialization.NoEncryption(),
+        config_dict = config.serviceConnection.__root__.config.dict()
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/snowflake.py:68]: let's try to not convert to `dict`if possible. `connectionArguments` is an attribute of `SnowflakeConnection` which has `None` as default, so we should be good to go with:

```
connection_arguments = config.serviceConnection.__root__.config.connectionArguments
if connection_arguments:
    ...
```

**Review by pmbrull — CHANGES_REQUESTED** [review]
Thanks for the PR, small comment to maintain a consistent style

File: ingestion/src/metadata/ingestion/source/snowflake.py [hunk 2]
```diff
@@ -89,12 +100,12 @@ def get_databases(self) -> Iterable[Inspector]:
...
                 self.connection.execute(use_db_query)
                 logger.info(f"Ingesting from database: {row[1]}")
-                self.config.database = row[1]
+                self.config.serviceConnection.__root__.config.database = row[1]
                 self.engine = get_engine(self.config)
                 yield inspect(self.engine)
 
     def get_table_fqn(self, service_name, schema, table_name) -> str:
-        return f"{service_name}{FQDN_SEPARATOR}{self.config.database}{FQDN_SEPARATOR}{schema}{FQDN_SEPARATOR}{table_name}"
+        return f"{service_name}{FQDN_SEPARATOR}{self.config.serviceConnection.__root__.config.database}{FQDN_SEPARATOR}{schema}{FQDN_SEPARATOR}{table_name}"
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/snowflake.py:108]: Let's remove the FQN computation on the source. Not needed for the backend.

File: ingestion/src/metadata/ingestion/source/sql_source.py [hunk 3]
```diff
@@ -251,6 +251,15 @@ def next_record(self) -> Iterable[Entity]:
 
                 if self.source_config.includeTables:
                     yield from self.fetch_tables(inspector, schema)
+
+                if self.source_config.schemaFilterPattern:
+                    if self.source_config.schemaFilterPattern.includes:
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/sql_source.py:256]: Use the filter module here https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/utils/filters.py

Comment (pmbrull) [reply ingestion/src/metadata/ingestion/source/sql_source.py:256]: nvm, all this addition can be removed. It is done at line 246!

**Review by pmbrull — CHANGES_REQUESTED** [review]

**Review by pmbrull — APPROVED** [review]
Thanks for the changes. LGTM. Let's merge it in after checking the conflicts

**Comment by ayush-shah** [comment]
@codingwithabhi prioritize this connector and let's try to merge it in asap

**Pull request merged**